includes/class-sensei-context-notices.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * File containing Sensei_Context_Notices class.
+ *
+ * @package sensei-lms
+ * @since 3.15.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei_Context_Notices class.
+ *
+ * @since 3.15.0
+ */
+class Sensei_Context_Notices {
+	/**
+	 * Notices context.
+	 *
+	 * @var string
+	 */
+	private $context;
+
+	/**
+	 * Notices instances.
+	 *
+	 * @var self[]
+	 */
+	private static $instances = [];
+
+	/**
+	 * Array with notices to show in the specific context.
+	 *
+	 * @var array
+	 */
+	private $notices = [];
+
+	/**
+	 * CSS class prefix.
+	 *
+	 * @var string
+	 */
+	private $css_class_prefix = 'sensei-lms';
+
+	/**
+	 * Sensei_Context_Notices constructor. Prevents other instances from being created outside of `self::instance()`.
+	 *
+	 * @param string $context          The context of the notices.
+	 * @param string $css_class_prefix CSS class prefix to be applied in the HTML.
+	 */
+	private function __construct( string $context, string $css_class_prefix = null ) {
+		$this->context             = $context;
+		$this->notices[ $context ] = [];
```

Comment:
I don't understand the purpose of having the `$notices` variable indexed by `$context`.
I see that we will have one instance per `$context` and looks like every instance will only have its corresponding notices so, why having also `$this->notices[ $context ] = [];` instead of just `$this->notices = [];` ?

Reply:
Ah, makes total sense! Changed here: https://github.com/Automattic/sensei/pull/4477/commits/f4efa9f02d3cc72ec00bbabb0bbfe029ca983d09